src/pkg/utils/io.go

```diff
@@ -422,3 +424,64 @@ func SHAsMatch(path, expected string) error {
 	}
 	return nil
 }
+
+// CreateReproducibleTarballFromDir creates a tarball from a directory with stripped headers
+func CreateReproducibleTarballFromDir(dirPath, dirPrefix, tarballPath string) error {
```

Comment:
Can we check this real fast to ensure that this works cross platform (should be easy in a <20 test to have a package create run, then unarchive and check the component / sbom tarballs, then create againa nd check again)